runtime/src/validate_block/implementation.rs

```diff
@@ -220,6 +233,20 @@ impl<B: BlockT> Storage for WitnessStorage<B> {
 			self.overlay.insert(key, None);
 		}
 	}
+
+	fn storage_append(&mut self, key: &[u8], value: Vec<u8>) {
+		let value_vec = sp_std::vec![EncodeOpaqueValue(value)];
+		let current_value = self.overlay.entry(key.to_vec()).or_default();
+
+		if let Some(item) = sp_std::mem::take(current_value) {
+			*current_value = Some(match Vec::<EncodeOpaqueValue>::append_or_new(item, &value_vec) {
+				Ok(item) => item,
+				Err(_) => value_vec.encode(),
+			});
+		} else {
+			*current_value = Some(value_vec.encode());
+		}
+	}
```

Comment:
Implementation with the entry api instead of remove/insert. Is it good? :sweat_smile: I guess since it doesn't remove and then re-insert, this saves an allocation